testutils/src/main/java/com/amplifyframework/testutils/VoidResult.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.testutils;
+
+/**
+ * Result placeholder to be used when there is no value emitted from a task, but we are interested in the completion of
+ * the task.
+ */
+public final class VoidResult {
```

Comment:
Nice!! I think there are a few places where I could have used this, around the code base. I can go back and patch them up, now that you've added this.